spec/controllers/products_controller_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+  describe "GET #index" do
+
+    it "loads just active products into @products" do
+      product1 = Product.create(:name => "1st Product Name", :price => 2, :user_id => 3, :status => "active")
+      product2 =  Product.create(:name => "2nd Product Name", :price => 5, :user_id => 4, :status => "retired")
+      get :index
+      expect(assigns(:products)).to match_array([product1])
+    end
+  end
+
+  describe "GET#New" do
+    it "saves a new blank instance of product in a variable" do
+      @product = Product.new(id: 1, name: "some name", price: 1, user_id: 1, status: "active")
+      @product.save
+
+      get :new
+      expect(Product.count).to eq 1
+    end
+
+    it "saves a new blank instance of review in a variable" do
+      @product = Product.new(id: 7, name: "some name", price: 1, user_id: 1)
```

Comment:
You could just use the `create` method for both this and the `@review` which will reduce the lines of code here which would be good